packages/ra-ui-materialui/src/layout/MenuItemLink.js

```diff
@@ -1,61 +1,49 @@
-import React, { Component, cloneElement } from 'react';
+import React, { cloneElement } from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import { NavLink } from 'react-router-dom';
 import MenuItem from '@material-ui/core/MenuItem';
 import ListItemIcon from '@material-ui/core/ListItemIcon';
 import Tooltip from '@material-ui/core/Tooltip';
-import { withStyles, createStyles } from '@material-ui/core/styles';
+import { makeStyles } from '@material-ui/core/styles';
 
 const NavLinkRef = React.forwardRef((props, ref) => (
     <NavLink innerRef={ref} {...props} />
 ));
 
-const styles = theme =>
-    createStyles({
-        root: {
-            color: theme.palette.text.secondary,
-        },
-        active: {
-            color: theme.palette.text.primary,
-        },
-        icon: { minWidth: theme.spacing(5) },
-    });
+const useStyles = makeStyles(theme => ({
+    root: {
+        color: theme.palette.text.secondary,
+    },
+    active: {
+        color: theme.palette.text.primary,
+    },
+    icon: { minWidth: theme.spacing(5) },
+}));
 
-export class MenuItemLink extends Component {
-    static propTypes = {
-        classes: PropTypes.object.isRequired,
-        className: PropTypes.string,
-        leftIcon: PropTypes.element,
-        onClick: PropTypes.func,
-        primaryText: PropTypes.node,
-        staticContext: PropTypes.object,
-        to: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
-            .isRequired,
-    };
+function MenuItemLink({
+    classes: classesOverride,
+    className,
+    primaryText,
+    leftIcon,
+    onClick,
+    sidebarIsOpen,
+    ...props
+}) {
+    const classes = useStyles({ classes: classesOverride });
 
-    handleMenuTap = e => {
-        this.props.onClick && this.props.onClick(e);
+    const handleMenuTap = e => {
```

Comment:
this one should be memoized with `useCallback`

Reply:
Sure, I've done this. Thanks for looking at it!